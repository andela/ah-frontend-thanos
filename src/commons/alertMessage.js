const alertMessage = {
  title: 'Problem encountered',
  text: '',
  type: 'error',
  confirmButtonText: 'OK',
};

export default alertMessage;
