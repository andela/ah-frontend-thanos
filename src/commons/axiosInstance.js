import axios from 'axios';

const baseURL = 'https://ah-backend-thanos-stagin-pr-48.herokuapp.com/';
const timeout = 60000;
const token = localStorage.getItem('token');
const headers = token
  ? {
    'Content-Type': 'application/json',
    Authorization: `Token ${token}`,
  } : {
    'Content-Type': 'application/json',
  };
const axiosInstance = axios.create({
  baseURL,
  timeout,
  headers,
});
export default axiosInstance;
