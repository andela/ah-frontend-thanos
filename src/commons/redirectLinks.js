// redirect links at the bottom of signup and login pages
import React from 'react';

const generateRedirectLinks = (message, link, linkName) => (
  <div className="small mt-3">
    <span>{message}</span>
    <a href={`/${link}`}>{linkName}</a>
  </div>
);

export default generateRedirectLinks;
