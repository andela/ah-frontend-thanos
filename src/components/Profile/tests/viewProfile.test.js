import React from 'react';
import { shallow, mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import ProfileConnected from '../../../containers/profiles/profiles';
import ViewProfile from '../viewProfile';
import { sampleProfiles } from '../../../commons/initialStates';

describe('Profile component', () => {
  let wrapper;
  const mockStore = configureMockStore();
  const getProfileAction = jest.fn();
  const nextProps = {
    isLoggedIn: true,
    profilePayload: { first_name: 'rachael', last_name: 'Zeus', bio: 'this is my bio' },
  };
  const props = {
    history: { push: jest.fn() },
  };
  const match = {
    params: { username: 'rachaelminani' },
  };
  beforeEach(() => {
    const store = mockStore({});
    wrapper = shallow(
      <Provider store={store}>
        <ProfileConnected
          {...props}
          getProfileAction={getProfileAction}
          followersList={sampleProfiles}
          followeesList={sampleProfiles}
          match={match}
        />
      </Provider>,
    );
    wrapper.setState({});
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not redirect if isLoggedIn is false', () => {
    wrapper.setProps({ ...nextProps });
    expect(props.history.push).toBeCalledTimes(0);
  });

  it('renders ViewProfile', () => {
    const viewProfile = mount(
      <ViewProfile followersList={sampleProfiles} followeesList={sampleProfiles} />,
    );
    expect(viewProfile).toMatchSnapshot();
  });
});
