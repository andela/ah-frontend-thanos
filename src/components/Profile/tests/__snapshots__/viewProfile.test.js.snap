// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should render correctly 1`] = `
<Connect(Profile)
  getProfileAction={[MockFunction]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
  match={
    Object {
      "params": Object {
        "username": "rachaelminani",
      },
    }
  }
/>
`;
